web/src/filter.ts

```diff
@@ -31,11 +73,17 @@ function zephyr_stream_name_match(message, operand) {
     return related_regexp.test(stream_name);
 }
 
-function zephyr_topic_name_match(message, operand) {
+function zephyr_topic_name_match(message: Message, operand: string): boolean {
+    if (message.type !== "stream") {
+        return false;
+    }
     // Zephyr users expect narrowing to topic "foo" to also show messages to /^foo(.d)*$/
     // (foo, foo.d, foo.d.d, etc)
     // TODO: hoist the regex compiling out of the closure
     const m = /^(.*?)(?:\.d)*$/i.exec(operand);
+    if (m === null) {
+        return false;
```

Comment:
This is creating codecov issues. Is it reasonable to expect it to always get a match? It seems like it has thus far without throwing errors

Reply:
This clearly matches all strings; we can use `assert(m !== null)` or (with a comment) `!`.